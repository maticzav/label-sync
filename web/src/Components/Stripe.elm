module Components.Stripe exposing (stripe)

import Element exposing (..)
import Element.Background as Background
import Element.Border as Border
import Element.Font as Font
import Html.Attributes as Attr


labels : List ( String, Element.Color )
labels =
    [ ( "bug/0-needs-info", rgb255 238 0 0 )
    , ( "area/database", rgb255 251 202 4 )
    , ( "kind/improvement", rgb255 59 91 219 )
    , ( "bug/2-confirmed", rgb255 238 0 0 )
    , ( "kind/bug", rgb255 59 91 219 )
    , ( "area/docs", rgb255 251 202 4 )
    , ( "kind/feature", rgb255 59 91 219 )
    , ( "status/stale", rgb 36 218 141 )
    ]


stripe : Element msg
stripe =
    row
        [ width (fill |> maximum 700)
        , spacing 15
        , paddingXY 0 10
        , centerX
        , Element.htmlAttribute <| Attr.id "stripe"
        ]
        (List.map viewLabel labels)


viewLabel : ( String, Element.Color ) -> Element msg
viewLabel ( label, color ) =
    el
        [ Background.color color
        , Border.rounded 3
        , Font.color (rgb255 255 255 255)
        , Font.size 14
        , Font.bold
        , paddingXY 5 3
        , Font.family [ Font.sansSerif ]
        , Border.innerShadow
            { offset = ( 0.0, -0.4 )
            , size = 0.0
            , blur = 2.0
            , color = rgb255 27 31 35
            }
        ]
        (text label)
