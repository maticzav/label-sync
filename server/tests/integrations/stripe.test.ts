import { PrismaClient, Plan } from '@prisma/client'
import { Timber } from '@timberio/node'
import execa from 'execa'
import nock from 'nock'
import { Probot, createProbot } from 'probot'
import Stripe from 'stripe'

const manager = require('../../src')

describe('stripe:', () => {
  let prisma: PrismaClient
  let timber: Timber
  let stripe: Stripe

  let timberLogs: string[] = []

  beforeAll(async () => {
    // Network settings
    nock.disableNetConnect()
    // local servers
    nock.enableNetConnect((host) => {
      return (
        host.includes('localhost') ||
        host.includes('127.0.0.1') ||
        host.includes('stripe')
      )
    })

    // DataStores
    prisma = new PrismaClient()
    timber = new Timber('apikey', 'source', {
      ignoreExceptions: false,
    })
    stripe = new Stripe(process.env.STRIPE_API_KEY!, {
      apiVersion: '2020-03-02',
    })

    /* Setup probot */
    probot = createProbot({
      id: 1,
      cert: 'test',
      githubToken: 'test',
      port: 5050,
    })

    probot.load((app) => manager(app, prisma, timber, stripe))
    probot.start()
  })

  afterAll(async () => {
    nock.cleanAll()
    nock.enableNetConnect()

    probot.httpServer!.close()
    await prisma.disconnect()
  })

  let probot: Probot

  beforeEach(async () => {
    /* Clean the database */
    await prisma.installation.deleteMany({ where: {} })

    /* Mock all timber logs */
    nock('https://logs.timber.io')
      .post(/.*/)
      .reply(200, (uri, body) => {
        /* remove timestamp from log */
        const [log] = body as any
        delete log['dt']
        delete log['periodEndsAt']
        /* collect logs */
        timberLogs.push(JSON.stringify(log))
        return
      })
      .persist()
  })

  afterEach(() => {
    nock.cleanAll()
  })

  /* Tests */

  test.skip('checkout.session.completed', async () => {
    execa('stripe', ['listen', '--forward-to localhost:5050/stripe'], {
      stdio: 'inherit',
      detached: true,
      cleanup: true,
    })
    await execa('stripe', ['trigger', 'checkout.session.completed'])

    /* Test database */

    const installation = await prisma.installation.findOne({
      where: { account: 'maticzav' },
    })

    delete installation!.createdAt
    delete installation!.updatedAt
    delete installation!.periodEndsAt
    expect(installation).toMatchSnapshot()
  })

  test.skip('logs are reporting correctly', async () => {
    expect(timberLogs).toMatchSnapshot()
  })

  /* End of tests */
})
