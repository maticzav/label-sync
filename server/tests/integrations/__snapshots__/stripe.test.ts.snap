// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stripe: logs are reporting correctly 1`] = `Array []`;
